viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    movie = {}
+    movie_set = {title, genre, rating}
+
+    if None in movie_set:
+        return None
+    else:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title != movie["title"]:
+            continue
+        else:
+            movie_index = user_data["watchlist"].index(movie)
+            removed_movie = user_data["watchlist"].pop(movie_index)
+            add_to_watched(user_data, removed_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    total = 0
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    
+    average = total / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+    
+    compare_dict = {}
+
+    if not user_data["watched"]:
+        return None
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre not in compare_dict:
+            compare_dict[genre] = 0
+        compare_dict[genre] += 1
+    
+    most_watched_genre = max(compare_dict, key=compare_dict.get)

```

Comment:
Nice use of `max()` to find the genre with the max count. How would you do something similar if a function like `max()` didn't exist? And what does is the `key=compare_dict.get` portion doing? Now is a great time to think through how library functions solve these problems!

Reply:
**Notes for Myself:**

`key=compare_dict.get` is an argument passed into the max function to return the key with the highest value in the dictionary

If we didn't have the max function, the code would look something like:

```
popular_genre_count = 0
most_watched_genre = None
for genre, count in compare_dict.items():
    if count > popular_genre_count:
        popular_genre_count = count
        most_watched_genre = genre

return most_watched_genre
```